extensions/sns/e2e/tests/sns.bash

```diff
@@ -93,7 +98,44 @@ SNS_CONFIG_FILE_NAME="sns.yml"
     #dfx canister id sns_swap
 }
 
-@test "sns-cli binary exists and is executable" {
-    run "$(dfx cache show)"/extensions/sns/sns-cli --help
-    assert_output --partial "Initialize, deploy and interact with an SNS"
+# This test asserts that the `prepare-canisters` subcommand and it's child subcommands
+# exist in the current extension version.
+@test "sns prepare-canisters exists" {
+    run dfx sns prepare-canisters --help
+    assert_output --partial "dfx sns prepare-canisters"
+    run dfx sns prepare-canisters add-nns-root --help
+    assert_output --partial "dfx sns prepare-canisters add-nns-root"
+    run dfx sns prepare-canisters remove-nns-root --help
+    assert_output --partial "dfx sns prepare-canisters remove-nns-root"
+}
+
+# This test asserts that the new subcommand `prepare-canister add-nns-root` can add NNS root
+# as a co-controller to a dapp.
+@test "sns prepare-canisters adds NNS Root" {
+     dfx_extension_install_manually nns
+     install_shared_asset subnet_type/shared_network_settings/system
+     dfx start --clean --background --host 127.0.0.1:8080
+     sleep 1
```

Comment:
What's `sleep 1` for here?  `dfx start --background` waits until the replica reports `healthy` status, but you can double-check here (the following calls `dfx ping --wait-healthy`)
```suggestion
     wait_until_replica_healthy
```

Reply:
I was following the structure of an already created test and didn't know this function existed. I've upgrade all three tests to use this now. Thanks for the pointer!